preprocessors/multistage-textbook-diagram.schema.json

```diff
@@ -0,0 +1,149 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://image.a11y.mcgill.ca/preprocessors/mutistage-textbook-diagram.schema.json",
+    "type": "object",
+    "title": "Multistage Textbook Diagram Data",
+    "description": "JSON containing pixel coordinates of different stages in a mutistage textbook diagram and the links between them",
+    "definitions": {
+        "normCoord": {
+            "description": "A pair of normalized coordinates.",
+            "type": "array",
+            "items": {
+                "type": "number",
+                "minimum": 0,
+                "maximum": 1
+            },
+            "minItems": 2,
+            "maxItems": 2
+        },
+        "contour": {
+            "type": "object",
+            "description": "A single contour of a segment as detected. Each contour is a separate region.",
+            "properties": {
+                "coordinates": {
+                    "description": "The normalized coordinates forming the contour as returned by OpenCV.",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/normCoord"
+                    }
+                },
+                "centroid": {
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/normCoord"
+                        },
+                        {
+                            "description": "The center of the contour with normalized coordinates."
+                        }
+                    ]
+                },
+                "area": {
+                    "description": "The area of the image taken by this contour",
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 1
+                }
+            }
+        }
+    },
+    "properties": {
+        "nodes": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "description": "A stage in the flow diagram",
```

Comment:
Is this for flow diagrams? Other textbook diagrams?

Reply:
This is intended for multiple stages, e.g., cycles/ flow diagrams.

Reply:
Is there a taxonomy of diagrams you're using?

Reply:
I don't know if these are established (or could be termed 'taxonomy') but I have been treating 'multistage diagrams' as equivalent to 'network charts' defined in this paper (https://doi.org/10.1007/s11165-022-10058-6). 
_"A network chart depicts abstract relationships among different visual components, such as flow chart or decision tree (LaDue et al., 2015). Although they are sometimes classified and labelled as diagrams (e.g. Venn diagram), network charts are different from diagrams because they represent ideas, processes or relationships without any direct correspondence to a real physical object."_
[LaDue et al.](https://doi.org/10.1007/s10956-015-9566-4) seems to call this diagram type "Network"